server/protocols/webadmin/webadmin-mailqueue/src/test/java/org/apache/james/webadmin/routes/MailQueueRoutesTest.java

```diff
@@ -477,55 +503,90 @@ public void forcingDelayedMailsDeliveryShouldRejectRequestWithoutBody() throws E
     @Test
     public void deleteMailsShouldDeleteMailsWhenSenderIsGiven() throws Exception {
         MemoryMailQueue queue = mailQueueFactory.createQueue(FIRST_QUEUE);
-        String sender = "sender@james.org";
-        queue.enQueue(Mails.defaultMail()
-                .sender(sender)
-                .build());
 
-        given()
-            .param("sender", sender)
-        .when()
+        queue.enQueue(FakeMail.builder()
+            .name(FAKE_MAIL_NAME_1)
+            .sender(SENDER_1_JAMES_ORG)
+            .build());
+
+        queue.enQueue(FakeMail.builder()
+            .name(FAKE_MAIL_NAME_2)
+            .sender(SENDER_2_JAMES_ORG)
+            .build());
+
+        String taskId = with()
+            .param("sender", SENDER_1_JAMES_ORG)
             .delete(FIRST_QUEUE + "/mails")
-        .then()
-            .statusCode(HttpStatus.NO_CONTENT_204);
+            .jsonPath()
+            .getString("taskId");
+
+        taskShouldCompleteAndHasDetail(taskId, 2, 1);
+    }
 
-        assertThat(queue.browse()).isEmpty();
+    private void taskShouldCompleteAndHasDetail(String taskId, int initialCount, int remainingCount) {
+        given()
+            .basePath(TasksRoutes.BASE)
+            .when()
```

Comment:
We have a convention for indenting RestAssured calls:

given, when then with are on the same line.

Here you should have:

```
given()
    .basePath(TasksRoutes.BASE)
.when()
    .get(taskId + "/await")
.then()
    .body("status", is("completed"))
    ...
```